src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -478,3 +478,101 @@ function CircularLadderGraph(n::T) where {T <: Integer}
     add_edge!(g, n+1, 2*n)
     return g
 end
+
+"""
+    BarbellGraph(n)
+
+Create a [barbell graph](https://en.wikipedia.org/wiki/Barbell_graph) consisting of a clique of size `n1` connected by an edge to a clique of size `n2`.
+
+### Implementation Notes
+Preserves the eltype of `n1` and `n2`. Will error if the required number of vertices
+exceeds the eltype.
+`n1` and `n2` must be at least 1 so that both cliques are non-empty.
+"""
+function BarbellGraph(n1::T, n2::T) where {T <: Integer}
+    (n1 < 1 || n2 < 1) && throw(DomainError("n1=$n1 and n2=$n2 must be at least 1"))
+
+    if n1 == 1 && n2 == 1
+        return PathGraph(T(2))
+    elseif n1 > 1 && n2 == 1
+        g = CompleteGraph(n1)
+        add_vertex!(g)
+        add_edge!(g, n1, n1+1)
+        return g
+    elseif n1 == 1 && n2 > 1
+        g = blockdiag(PathGraph(T(1)), CompleteGraph(n2))
+        add_edge!(g, 1, 2)
+        return g
+    end
+
+    Tw = widen(T)
+    temp = T(Tw(n1)+Tw(n2)) # test to check if T is large enough
+
+    fadjlist = Vector{Vector{T}}(undef, n1+n2)
+    ne = Int(n1*(n1-1)/2 + n2*(n2-1)/2)
+
+    @inbounds @simd for u = 1:n1
+        listu = Vector{T}(undef, n1-1)
+        listu[1:(u-1)] = 1:(u - 1)
+        listu[u:(n1-1)] = (u + 1):n1
+        fadjlist[u] = listu
+    end
+
+    @inbounds for u = 1:n2
+        listu = Vector{T}(undef, n2-1)
+        listu[1:(u-1)] = (n1+1):(n1+(u-1))
```

Comment:
Not sure if it is better, but instead of writting `(n1+1):(n1+(u-1))` you can also write `(1:u-1) .+ n1`.

Reply:
Yeah, I originally had something like this but I wasn't sure what the preferred style was.

Reply:
Why couldn't we just create two `CompleteGraph`s of sizes `n1` and `n2`, and then use `blockdiag` and a single `add_edge`? If this is more efficient, perhaps we can move that efficiency to these other building block functions instead.

Reply:
But `blockdiag` is not an efficient operator. And even if it where, that would mean that we temporarily allocate twice the necessary memory.